pkgs/development/compilers/llvm/10/compiler-rt/gnu-install-dirs.patch

```diff
@@ -19,7 +19,7 @@ index 35a48c6af29c..e4300f256091 100644
    # Install in Clang resource directory.
    install(FILES ${file_name}
 -    DESTINATION ${COMPILER_RT_INSTALL_PATH}/share
-+    DESTINATION ${COMPILER_RT_INSTALL_PATH}/${CMAKE_INSTALL_FULL_INCLUDEDIR}
++    DESTINATION ${COMPILER_RT_INSTALL_PATH}/${CMAKE_INSTALL_FULL_DATADIR}
```

Comment:
As @primeos [pointed out](https://github.com/NixOS/nixpkgs/pull/122068#issuecomment-841653420), `${COMPILER_RT_INSTALL_PATH}` has to be empty, so why not get rid of it completely?
```suggestion
+    DESTINATION ${CMAKE_INSTALL_FULL_DATADIR}
```
Also see https://github.com/jtojnar/cmake-snips#assuming-cmake_install_dir-is-relative-path.

Reply:
I'm not sure why that choice was made (could have reasons, e.g. if there's a similar upstream patch but combining those paths without something like `get_filename_component()` seems unlikely) but it affects all of `gnu-install-dirs.patch` and that should be done in a separate PR IMO / is out-of-scope here (there are much more pressing concerns atm).

Reply:
it's no longer super pressing, but we should probably do the right thing now.

Reply:
I'm not quite sure. Doing the right thing would mean touching the `gnu-install-dirs.patch` for every llvm related package we have currently which means 60 patches would need to be touched and reviewed 😐

I personally don't think I'll have the time to do this especially not in time for the next staging-next PR.

Reply:
Ah, nevermind then :)

Reply:
I wonder what the state of the upstream review of the patch is, @Ericson2314. I guess LLVM merging the patch would be a good point in time to make another effort of backporting the patch to the different LLVM versions we have.